.gitignore

```diff
@@ -1,18 +1,19 @@
 # Build directories and binary files
-build/
-out/
-cmake-build-*/
+**/build/
+**/out/
+**/cmake-build-*/
 
 # IDE files
-.vs/
-.idea/
-.vscode/
+**/.vs/
+**/.idea/
+**/.vscode/
```

Comment:
I am not clear on why this change is being made. The commit message just says "ignore folders recursively". Is there a reason for adding this? Are there folders that VS Code adds to your repo all over your project, rather than just the root level?

Keep in mind that this is a general-purpose best-practices template project. Users are expected to add new items to the `.gitignore` as needed. The default `.gitignore` is meant to provide sane defaults that anyone can use, with minimal impact on anyone that hasn't read it yet. With this change, I could add a `src/util/build/` directory full of cpp files, and if I do not carefully read my `.gitignore`, I would not be able to commit anything in the directory. That would be a nasty surprise.

Reply:
I accidentally opened the src/qt with visual studio using the open folder functionality and the VS created the .vs folder under src/qt/.vs. It was just an accident. I know that only the root folder should be the entry point since the main CMakeLists is there. I can revert that I do not consider this change important